File: src/main/cpp/src/synapse/biological_somatic_spine.cc
#include "synapse/biological_somatic_spine.h"

#include "synapse/physical_somatic_spine.h"

namespace cx3d {
namespace synapse {

std::shared_ptr<BiologicalSomaticSpine> BiologicalSomaticSpine::create() {
  return std::shared_ptr<BiologicalSomaticSpine>(new BiologicalSomaticSpine());
}

BiologicalSomaticSpine::BiologicalSomaticSpine() {
}

BiologicalSomaticSpine::~BiologicalSomaticSpine() {
}

bool BiologicalSomaticSpine::equalTo(const std::shared_ptr<BiologicalSomaticSpine>& other) const {
  return this == other.get();
}

StringBuilder& BiologicalSomaticSpine::simStateToJson(StringBuilder& sb) const {
  sb.append("{}");
  return sb;
}

std::shared_ptr<PhysicalSomaticSpine> BiologicalSomaticSpine::getPhysicalSomaticSpine() const {
  return physical_somatic_spine_;
}

void BiologicalSomaticSpine::setPhysicalSomaticSpine(const std::shared_ptr<PhysicalSomaticSpine>& ps) {
  physical_somatic_spine_ = ps;
}

}  // namespace synapse
}  // namespace cx3d

File: src/main/java/ini/cx3d/synapses/interfaces/BiologicalSomaticSpine.java
package ini.cx3d.synapses.interfaces;

import ini.cx3d.SimStateSerializable;

/**
 * Created by lukas on 18.04.16.
 */
public interface BiologicalSomaticSpine extends SimStateSerializable {
    @Override
    ini.cx3d.swig.NativeStringBuilder simStateToJson(ini.cx3d.swig.NativeStringBuilder sb);

    ini.cx3d.synapses.interfaces.PhysicalSomaticSpine getPhysicalSomaticSpine();

    void setPhysicalSomaticSpine(ini.cx3d.synapses.interfaces.PhysicalSomaticSpine physicalSomaticSpine);
}
